elm-format-lib/src/ElmFormat/Render/Box.hs

```diff
@@ -1944,15 +1944,29 @@ formatString elmVersion style s =
     stringBox quotes escaper =
       line $ row
           [ punc quotes
-          , literal $ escaper $ concatMap fix s
+          , literal $ escaper $ comonadic_ish_map fix s
```

Comment:
As you noted, we should look at doing this in the parse step so that the AST does not contain the CR's

Reply:
b106094 reverts this commit.

356ce1b adds this feature to the parse step instead. When a CR is seen the string parsers look ahead and checks if there's an LF following. If so, it behaves as it would if encountering just an LF, but the pointer is incremented by two instead of one.